app/src/main/java/org/openobservatory/ooniprobe/domain/MeasurementsManager.java

```diff
@@ -109,6 +117,57 @@ public String getReadableEntry(Measurement measurement) throws IOException {
         return jsonPrinter.prettyText(FileUtils.readFileToString(entryFile, StandardCharsets.UTF_8));
     }
 
+    public void syncMeasurements(Activity activity,
+                                 OnStartResubmission onStartResubmission,
+                                 OnCancelResubmission onCancelResubmission,
+                                 Integer... params) {
+
+        if (determineIfMeasurementsWereTakenOverVPN(params)) {
+            new AlertDialog.Builder(activity, R.style.MaterialAlertDialogCustom)
+                    .setTitle(activity.getString(R.string.Modal_UploadVPNResults_Title))
+                    .setMessage(activity.getString(R.string.Modal_UploadVPNResults_Message))
+                    .setNegativeButton(R.string.Modal_Cancel, (dialogInterface, i) -> {
+                        dialogInterface.dismiss();
+                        onCancelResubmission.onCancel();
+                    })
+                    .setPositiveButton(R.string.Modal_OK, (dialogInterface, i) -> {
+                        onStartResubmission.onResubmit();
+                    })
+                    .show();
+        } else {
+            onStartResubmission.onResubmit();
+        }
+    }
+
+    public boolean determineIfMeasurementsWereTakenOverVPN(Integer... params) {
+        if (params.length != 2)
+            throw new IllegalArgumentException("MKCollectorResubmitTask requires 2 nullable params: result_id, measurement_id");
+        Where<Measurement> msmQuery = Measurement.selectUploadable();
+        if (params[0] != null) {
+            msmQuery.and(Measurement_Table.result_id.eq(params[0]));
+        }
+        if (params[1] != null) {
+            msmQuery.and(Measurement_Table.id.eq(params[1]));
+        }
+        //Get a list of measurements with report file
+        List<Measurement> measurements = Measurement.withReport(context, msmQuery);
+
+        return Lists.transform(measurements, measurement -> {
+            try {
+                return ((Application) context.getApplicationContext()).getGson().fromJson(
+                        FileUtils.readFileToString(
+                                Measurement.getEntryFile(context, measurement.id, measurement.test_name),
+                                StandardCharsets.UTF_8
+                        ),
+                        Measurement.DataRoot.class
```

Comment:
I am not getting why `DataRoot` is named as such? Is it named as such because it's the toplevel object representing a measurement?